src/formatDate.js

```diff
@@ -8,7 +8,54 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [oldData, idFromYear, oldYear, fromSymbol] = getFormat(fromFormat);
+  const [newData, , newYear, toSymbol] = getFormat(toFormat);
+
+  const fromDate = date.split(fromSymbol);
+  const toDate = [];
+
+  const year = fromDate[idFromYear];
+
+  let formatYear = '';
+
+  if (oldYear !== newYear) {
+    if (newYear < oldYear) {
+      formatYear += year.slice(2);
+    }
+
+    if (newYear > oldYear) {
+      if (year >= 30) {
+        formatYear += '19' + year;
+      } else {
+        formatYear += '20' + year;
```

Comment:
The logic for determining the century (lines 27-30) assumes that any year >= 30 belongs to the 20th century and any year < 30 belongs to the 21st century. This might not be accurate for all use cases, especially if dealing with historical dates or future dates beyond 2030. Consider adding a more flexible approach or documenting this assumption clearly.